lib/models/obms.js

```diff
@@ -0,0 +1,183 @@
+            if (key.match(secretsRegex)) {
+                obj.config[key] = encryption.decrypt(value);
+            }
+        });
+
+        return obj;
+    }
+
+    return Model.extend({
+        connection: 'mongo',
+        identity: 'obms',
+        attributes: {
+            node: {
+                model: 'nodes',
+                required: true
+            },
+            service: {
+                type: 'string',
+                required: true
+            },
+            config: {
+                type: 'json',
+                required: true,
+            },
+            toJSON: function() {
+                var obj = this.toObject();
+                return _deleteSecrets(obj);
+            }
+        },
+
+        beforeCreate: function(attrs, next) {
+            _encryptSecrets(attrs);
+            next();
+        },
+
+        beforeUpdate: function(attrs, next) {
+            if (attrs && attrs.config) {
+                _encryptSecrets(attrs);
+            }
+            next();
+        },
+
+        setIndexes: function() {
+            var indexes = [
+                {
+                    node: 1, service: 1
+                }
+            ];
+            return waterline.obms.createUniqueMongoIndexes(indexes);
+        },
+
+        findByNode: function(nodeId, service, reveal, query) {
+            var self = this;
+
+            query = query || {};
+            return waterline.nodes.findByIdentifier(nodeId)
+            .then(function(node) {
+                if (!node) { return node; }
+
+                _.merge(query, {service: service, node: node.id} );
+                return self.findOne(query);
+            })
+            .then(function(obm) {
+                if (obm && reveal) {
+                    return _revealSecrets(obm);
+                }
+                return obm;
+            })
+        },
+
+        // This function 'upserts' a new OBM into the obms collection.
+        // A few notes on this function's behavior:
+        // 1. This is not an isolated transaction.
+        // 2. There is a possibility that two clients could concurrently
+        //    attempt to upsert an OBM with the same node or service.
+        // 3. If note 2 occurs, mongo indexes will prevent the creation of
+        //    duplicate OBM resources.  It is the responsibility of the
+        //    caller to decide how to handle errors thrown as a result.
+        // 4. There is a possibility that another client has updated an OBM
+        //    between the atomic create or modify and find operations.  This
+        //    will cause the function fulfillment value to be inconsistent
+        //    with the provided obm arguments but consistent with the document
+        //    content at the time of the find query.
+        upsertByNode: function(nodeId, obm) {
+            var node;
+            var query;
+            var self = this;
+
+            return waterline.nodes.findByIdentifier(nodeId)
```

Comment:
Is this `findByIdentifier` call necessary? Seems like all that's being used is `node.id` which should be equal to `nodeId` anyways.